### PR TITLE
Removed some stale code related to --no-coordinator.

### PR DESCRIPTION
**Pull request by karya0**
Also removed fred_record_replay bits from jalloc.cpp.

### CODE REVIEW

File: src/processinfo.h [hunk 1]
```diff
@@ -204,7 +200,7 @@ class ProcessInfo
     uint32_t _numRestarts;
 
     uint32_t _numPeers;
-    uint32_t _noCoordinator;
+    uint32_t _dummy;
```
Comment (JainTwinkle) [inline src/processinfo.h:203]: Please add a small comment here about the dummy variable.

File: src/processinfo.h [hunk 2]
```diff
@@ -240,6 +216,29 @@ class ProcessInfo
     uint64_t _getcpu_offset;
     uint64_t _gettimeofday_offset;
     uint64_t _time_offset;
+
+    // Put <64-bit wide variabled here to ensure they are properly aligned.
```
Comment (JainTwinkle) [inline src/processinfo.h:220]: variabled -> variables

**Review by JainTwinkle — APPROVED** [review]
You can remove a stale FRed reference from the comment at [syscallreal.c:127](https://github.com/dmtcp/dmtcp/blob/9aea31b3dd974f5d2634b3161dfdd9844038bb63/src/syscallsreal.c#L127)
The rest looks good to me.

File: src/processinfo.h (hunk 2 above)
Comment (karya0) [reply src/processinfo.h:220]: Fixed

Comment (JainTwinkle) [reply src/processinfo.h:220]: Looks like you have merged the PR before pushing the typo fix. I guess you can take care of it in some other PR.

**Pull request merged**